E2E-ORS-Sample-For-Java/src/main/java/objectReplicationMonitor.java

```diff
@@ -0,0 +1,297 @@
+                            if (status.equals("complete")) {
+                                completed++;
+                                completedEvents.add(b);
+                            }
+
+                            // If the replication was failed on a specific blob, alert user and continue checking replication of other blobs
+                            else if (status.equals("failed")) {
+                                System.out.println("\nReplication of " + b + " has failed. Please check that " + b + " in destination container are not archived...");
+                                failedEvents.add(b);
+                                size--;
+                            }
+                        }
+                    }
+                }
+
+                // Check if status has a new value and percentage, to avoid repeat printing the same status
+                if ((double) completed / (double) size != ratio && (int) (((double) completed / (double) size) * 100) != percentage) {
+                    ratio = (double) completed / (double) size;
+                    percentage = (int) (ratio * 100);
+
+                    // Output completion status as a percentage
+                    System.out.println("\nReplication completion is at " + percentage + "%...");
+                    System.out.println(completed + " out of " + size + " blobs have successfully replicated...");
+
+                    // Check for completion of all blobs, this will also cause end of while loop
+                    if (ratio == 1) {
+                        System.out.println("\nCompleted!");
+                        break;
+                    }
+                }
+            }
+        }
+        return completedEvents;
+    }
+
+
+    private static void checkContents(String sourceConnectionString, String sourceContainer, String destinationConnectionString, String destinationContainer, List<String> blobList) throws FileNotFoundException {
+        /*
+        This method is to check that the contents of a replicated blob transferred correctly
+         */
+
+        String[] connectionString = {sourceConnectionString, destinationConnectionString};
+        String[] containerName = {sourceContainer, destinationContainer};
+        for (int s = 0; s < connectionString.length; s++) {
+
+            // Creating blobserviceclient to access containerclient
+            BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionString[s]).buildClient();
+
+            // Get the container client to access blob client
+            BlobContainerClient blobContainerClient = blobServiceClient.getBlobContainerClient(containerName[s]);
+
+            // Print out contents of one blob to show successful replication
+            BlobClient blobClient = blobContainerClient.getBlobClient(blobList.get(0));
+
+            System.out.println("\nContent of " + blobList.get(0) + " in the " + containerName[s] + " container:");
```

Comment:
So are we only comparing the first replicated blob?

Reply:
Yeah, right now all of the blobs have the same contents. I could have it compare for all of them, but if I'm using a high blob number like 1000+ then it would be printing out 2000 lines of "Hello World". I could do it for like a few blobs if you think that it is necessary though. I think I might need to edit my README slightly, because I am pretty sure that right now it is implying that all blobs will be compared